cmd/tink-worker/client/client_test.go

```diff
@@ -0,0 +1,222 @@
+	}()
+
+	dialer := func(ctx context.Context, _ string) (net.Conn, error) {
+		return listener.DialContext(ctx)
+	}
+
+	conn, err := ggrpc.DialContext(ctx, "bufnet", ggrpc.WithContextDialer(dialer), ggrpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("Failed to dial %s: %v", name, err)
+	}
+
+	client := workflow.NewWorkflowServiceClient(conn)
+	return client
+}
+
+func TestObtainTinkServerCreds(t *testing.T) {
+	t.Run("certURL arg validation", func(t *testing.T) {
+		expectedErrorMsg := "certURL cannot be empty"
+		_, err := ObtainTinkServerCreds("")
+		if err == nil {
+			t.Error("Error expected but none found")
+		} else if fmt.Sprint(err) != expectedErrorMsg {
+			t.Errorf("Error should be %v, got %v", expectedErrorMsg, err)
+		}
+	})
+
+	t.Run("ensure we get a response when fetching a cert", func(t *testing.T) {
+		expectedErrorMsg := "Unable to fetch cert from something"
+		_, err := ObtainTinkServerCreds("something")
+		if err == nil {
+			t.Error("Error expected but none found")
+		} else if fmt.Sprint(err) != expectedErrorMsg {
+			t.Errorf("Error should be %v, got %v", expectedErrorMsg, err)
+		}
+	})
+
+	t.Run("ensure an empty cert response is rejected", func(t *testing.T) {
+		emptyCert := ""
+		svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprint(w, emptyCert)
+		}))
+		defer svr.Close()
+
+		expectedErrorMsg := "Unable to parse the cert data"
+		_, err := ObtainTinkServerCreds(svr.URL)
+		if err == nil {
+			t.Error("Error expected but none found")
+		} else if fmt.Sprint(err) != expectedErrorMsg {
+			t.Errorf("Error should be %v, got %v", expectedErrorMsg, err)
+		}
+	})
+
+	t.Run("using with a self-signed cert", func(t *testing.T) {
+		cert, err := os.ReadFile(selfSignedCertFile)
+		if err != nil {
+			t.Errorf("Error opening self-signed cert file: %v", err)
+		}
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprintf(w, "%s\n", cert)
+		}))
+		defer server.Close()
+
+		_, err = ObtainTinkServerCreds(server.URL)
+		if err != nil {
+			t.Errorf("Error obtaining tink server creds: %v", err)
+		}
+	})
+}
+
+func TestEstablishServerConnection(t *testing.T) {
+	t.Run("grpcAuthority arg validation", func(t *testing.T) {
+		expectedErrorMsg := "grpcAuthority cannot be empty"
+		_, err := EstablishTinkServerConnection("", nil)
+		if err == nil {
+			t.Error("Error expected but none found")
+		} else if fmt.Sprint(err) != expectedErrorMsg {
+			t.Errorf("Error should be %v, got %v", expectedErrorMsg, err)
+		}
+	})
+
+	t.Run("attempt to connect to a bogus GRPC server", func(t *testing.T) {
+		expectedErrorMsg := "connect to tinkerbell server"
+		_, err := EstablishTinkServerConnection("bogus", nil)
+		if err == nil {
+			t.Error("Error expected but none found")
+		} else if fmt.Sprint(err) != expectedErrorMsg {
+			t.Errorf("Error should be %v, got %v", expectedErrorMsg, err)
+		}
+	})
+}
+
+func TestClientCreation(t *testing.T) {
```

Comment:
In this package, this workflow client only exists in this test package. In `main.go` we have `workflowClient := workflow.NewWorkflowServiceClient(conn)`. you might think about moving this test to a `main_test.go` that lives next to `main.go`

Reply:
Due to the change which moves these functions into `tink.go`, I think there will eventually be functions in here which deal with workflowClients, so I don't think this is necessary anymore?